.github/workflows/release_portable_linux_packages.yml

```diff
@@ -164,14 +164,31 @@ jobs:
       - name: Upload Releases to S3
         if: ${{ github.repository_owner == 'ROCm' }}
         run: |
-          # TODO: Append uploaded files to an index
           aws s3 cp ${{ env.DIST_ARCHIVE }} s3://therock-${{ env.RELEASE_TYPE }}-tarball/
           aws s3 cp ${{ env.OUTPUT_DIR }}/packages/dist/ s3://therock-${{ env.RELEASE_TYPE }}-python/${{ matrix.target_bundle.amdgpu_family }}/ \
           --recursive --no-follow-symlinks \
           --exclude "*" \
           --include "*.whl" \
           --include "*.tar.gz"
 
+      - name: Setup Python
+        if: ${{ github.repository_owner == 'ROCm' }}
+        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
+        with:
+          python-version: 3.12
+
+      - name: (Re-)Generate Python package release index
+        if: ${{ github.repository_owner == 'ROCm' }}
+        run: |
+          pip install boto3
+          ./build_tools/packaging/python/generate_release_index.py \
+            --bucket=therock-${{ env.RELEASE_TYPE }}-python \
+            --endpoint=s3.us-east-2.amazonaws.com \
+            --subdir=${{ matrix.target_bundle.amdgpu_family }} \
+            --output=${{ env.OUTPUT_DIR }}/packages/dist/index.html
+          # TODO: Integrate upload into `generate_release_index.py`
+          aws s3 cp ${{ env.OUTPUT_DIR }}/packages/dist/index.html s3://therock-${{ env.RELEASE_TYPE }}-python/${{ matrix.target_bundle.amdgpu_family }}/
```

Comment:
nit: could put the bucket name in a variable since it is reused between the generate release index script and the `aws s3 cp` command. Maybe future cleanup, moving all of the variable setup to the top of the workflow (or a script). That would make it easier to see how the various names like `therock-${{ env.RELEASE_TYPE }}-tarball/` and `therock-${{ env.RELEASE_TYPE }}-python` relate to one another